packages/hierarchies/src/hierarchies/internal/operators/ReleaseMainThreadOn.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { bufferCount, concatAll, concatMap, Observable } from "rxjs";
+import { LOGGING_NAMESPACE as commonLoggingNamespace } from "../Common";
+import { doLog } from "../LoggingUtils";
+
+const LOGGING_NAMESPACE = `${commonLoggingNamespace}.ReleaseMainThread`;
+
+/**
+ * Releases the main thread for other timers to use.
+ * @internal
+ */
+export const releaseMainThread = async () => {
+  doLog({ category: LOGGING_NAMESPACE, message: /* istanbul ignore next */ () => "Releasing main thread" });
+  return new Promise<void>((resolve) => setTimeout(resolve, 0));
+};
+
+/**
+ * Emits a certain amount of values, then releases the main thread for other timers to use.
+ * @internal
+ */
+export function releaseMainThreadOn<T>(elementCount: number) {
```

Comment:
Did you follow some convention coming up with this function name? I'd personally rename it to `releaseMainThreadOnItemsCount` or something like that.

Reply:
I was going with rxjs-like names but that sounds better, thanks.